### PR TITLE
Add option to download cpu profile for debugging.

### PR DESCRIPTION
**Pull request by kenzieschmoll**
Adds a button to the timeline page that, when clicked, downloads a json file with the current cpu profile response.

### CODE REVIEW

File: packages/devtools/lib/src/timeline/cpu_profile_protocol.dart [hunk 1]
```diff
@@ -12,13 +13,34 @@ import '../utils.dart';
 // debug and profile builds. Do they use different clocks and does this also
 // affect the sampling rate? See https://github.com/dart-lang/sdk/issues/36583.
 
+// Switch this flag to true collect debug info from the latest cpu profile. This
+// will also add a button to the timeline page that will download the debug info
```
Comment (jacob314) [inline packages/devtools/lib/src/timeline/cpu_profile_protocol.dart]: Nit: make this a /// comment and add a line break between the first and subsequent sentences.

File: packages/devtools/lib/src/timeline/cpu_profile_protocol.dart [hunk 2]
```diff
@@ -12,13 +13,34 @@ import '../utils.dart';
...
+// on click.
+const bool debugCpuProfile = false;
+
+/// Buffer that will store the latest cpu profile response json.
+///
+/// This buffer is for debug purposes. When [debugCpuProfile] is true, we will
+/// be able to dump this buffer to a downloadable json file.
+StringBuffer debugCpuProfileResponse = StringBuffer();
```
Comment (jacob314) [inline packages/devtools/lib/src/timeline/cpu_profile_protocol.dart]: nit: make this private and expose a getter that returns the string.

File: packages/devtools/lib/src/framework/framework_core.dart [hunk 3]
```diff
@@ -26,13 +28,35 @@ class FrameworkCore {
...
+    }
+    final queryParams = Uri.parse(window.location.toString()).queryParameters;
+
+    // TODO(kenzie): register a service on the devtools server that will dump
+    // all debug files behind the [debugTimeline] and [debugCpuProfile] flags.
+    // If we could automate this into some sort of bug report, we can probably
+    // remove this query param all together, but for now enabling manual dumps
+    // is fine.
+    final _debugTimeline = queryParams['debugTimeline'];
+    if (_debugTimeline != null && _debugTimeline == 'true') {
```
Comment (jacob314) [inline packages/devtools/lib/src/framework/framework_core.dart]: these != null checks are not needed.

Comment (jacob314) [reply packages/devtools/lib/src/framework/framework_core.dart]: simplify to
debugTimeline = queryParams['debugTimeline] == 'true';

File: packages/devtools/lib/src/timeline/cpu_profile_protocol.dart [hunk 4]
```diff
@@ -12,13 +13,37 @@ import '../utils.dart';
 // debug and profile builds. Do they use different clocks and does this also
 // affect the sampling rate? See https://github.com/dart-lang/sdk/issues/36583.
 
+/// Switch this flag to true collect debug info from the latest cpu profile.
+///
+/// This will add a button to the timeline page that will download the debug
+/// info on click.
+const bool debugCpuProfile = false;
```
Comment (jacob314) [inline packages/devtools/lib/src/timeline/cpu_profile_protocol.dart]: how can this be modified if it is const?

**Comment by jacob314** [comment]
lgtm

File: packages/devtools/lib/src/timeline/cpu_profile_protocol.dart (hunk 4 above)
Comment (kenzieschmoll) [reply packages/devtools/lib/src/timeline/cpu_profile_protocol.dart]: sorry, this was a staged-the-wrong-line-from-git-gui kind of a mistake.

File: packages/devtools/lib/src/framework/framework_core.dart (hunk 3 above)
Comment (kenzieschmoll) [reply packages/devtools/lib/src/framework/framework_core.dart]: done.

**Comment by kenzieschmoll** [comment]
@jacob314 is looking into the inspector travis failure. Should be unrelated to this change, so I'll go ahead and land this.

**Pull request merged**